src/dreyfus_index.erl

```diff
@@ -222,6 +243,35 @@ code_change(_OldVsn, State, _Extra) ->
 
 % private functions.
 
+get_index_with_sig(#doc{body={Fields}}=Doc, Sig) ->
```

Comment:
I didn't know about this case of index could be referred by multiple design documents. If that's the case I have to scan all the design docs in the db and ensure that it's not referred in other docs.